timbuctoo-instancev4/src/main/java/nl/knaw/huygens/timbuctoo/v5/graphql/mutations/MergeSchemaDataFetcher.java

```diff
@@ -84,12 +85,25 @@ public Object get(DataFetchingEnvironment env) {
     File customSchemaFile = fileHelper.fileInDataSet(ownerIdDataSetName.getLeft(), ownerIdDataSetName.getRight(),
       "customSchema.json");
 
+    Map<String, List<ExplicitField>> existingCustomSchema;
+
     try {
-      objectMapper.writeValue(customSchemaFile, customSchema);
+      existingCustomSchema = objectMapper.readValue(customSchemaFile,
```

Comment:
Add check that the file exists before trying to read the value. This prevents a lot of exceptions.